pkg/operator/controller.go

```diff
@@ -248,7 +271,11 @@ func (c *Controller) sync() error {
 		deploy = deploy.DeepCopy() // make sure we won't corrupt the cached vesrion
 	}
 
-	c.syncStatus(cr, deploy, applyError)
+	routes, err := c.getRoutes(cr)
+	if err != nil {
+		return err
```

Comment:
Failure to access one of the operator's routes shouldn't prevent the operator from reporting its status. Routes are expected to be absent when the operator is Removed (this blocks tests).

Reply:
done, ptal.